pkg/services/accesscontrol/acimpl/service.go

```diff
@@ -244,3 +248,102 @@ func (s *Service) DeclarePluginRoles(_ context.Context, ID, name string, regs []
 
 	return nil
 }
+
+// SearchUsersPermissions returns all users' permissions filtered by action prefixes
+func (s *Service) SearchUsersPermissions(ctx context.Context, user *user.SignedInUser, orgID int64,
+	options accesscontrol.SearchOptions) (map[int64][]accesscontrol.Permission, error) {
+	// Filter ram permissions
+	basicPermissions := map[string][]accesscontrol.Permission{}
+	for role, basicRole := range s.roles {
+		for i := range basicRole.Permissions {
+			if options.ActionPrefix != "" {
+				if strings.HasPrefix(basicRole.Permissions[i].Action, options.ActionPrefix) {
+					basicPermissions[role] = append(basicPermissions[role], basicRole.Permissions[i])
+				}
+			}
+			if options.Action != "" {
+				if basicRole.Permissions[i].Action == options.Action {
+					basicPermissions[role] = append(basicPermissions[role], basicRole.Permissions[i])
+				}
+			}
+		}
+	}
+
+	usersRoles, err := s.store.GetUsersBasicRoles(ctx, orgID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get managed permissions (DB)
+	usersPermissions, err := s.store.SearchUsersPermissions(ctx, orgID, options)
+	if err != nil {
+		return nil, err
+	}
+
+	// helper to filter out permissions the signed in users cannot see
```

Comment:
thought: should we request that the user is allowed to view all user's permissions to access this endpoint? It feels potentially dangerous to display partial permissions 🤔

Reply:
I'm going to keep this to be consistent with other endpoints, like `/api/org/users/search`